main.go

```diff
@@ -58,41 +64,62 @@ type customDNSProviderSolver struct {
 // You should not include sensitive information here. If credentials need to
 // be used by your provider here, you should reference a Kubernetes Secret
 // resource and fetch these credentials using a Kubernetes clientset.
-type customDNSProviderConfig struct {
-	// Change the two fields below according to the format of the configuration
-	// to be decoded.
+type ns1DNSProviderConfig struct {
 	// These fields will be set by users in the
 	// `issuer.spec.acme.dns01.providers.webhook.config` field.
 
-	//Email           string `json:"email"`
-	//APIKeySecretRef v1alpha1.SecretKeySelector `json:"apiKeySecretRef"`
+	APIKeySecretRef cmmeta.SecretKeySelector `json:"apiKeySecretRef"`
+	Endpoint        string                   `json:"endpoint"`
+	IgnoreSSL       bool                     `json:"ignoreSSL"`
+	TTL             int                      `json:"ttl"`
```

Comment:
inclined to remove TTL setting. Other providers have it, but since the DNS records this app creates are short lived, I don't see a use-case for having TTL be configurable

Reply:
did that